datalad/crawler/nodes/annex.py

```diff
@@ -336,7 +336,7 @@ def __init__(self, path=None,
         self._providers = Providers.from_config_files()
         self.yield_non_updated = yield_non_updated
 
-        if (not allow_dirty) and self.repo.dirty:
+        if (not allow_dirty) and self.repo.dirty():
```

Comment:
it is not about "size" ;)
regarding all the rest of arguments ("makes sense" etc) -- scrolling pages of `.dirty -> .dirty()` changes invalidates them. The only valid to me is somewhat of a nuisance to have both `is_dirty` and `dirty` but I can leave with that :-P